numpy/__init__.pyi

```diff
@@ -203,12 +202,12 @@ from typing import (
     TypeAlias,
 )
 
-if sys.version_info >= (3, 11):
-    from typing import LiteralString
-elif TYPE_CHECKING:
-    from typing_extensions import LiteralString
-else:
-    LiteralString: TypeAlias = str
+# NOTE: `typing_extensions` is always available in `.pyi` stubs or when
+# `TYPE_CHECKING` - even if not available at runtime.
+# This is because the `typeshed` stubs for the standard library include
+# `typing_extensions` stubs:
+# https://github.com/python/typeshed/blob/main/stdlib/typing_extensions.pyi
```

Comment:
Is this comment more of a "why did we make this change" statement?

Maybe it's better as part of the commit message.  Then it would be reachable from any of the changed lines via `git blame` or `git log -G"typing_extensions"`.

Reply:
Since it's not common knowledge that `typing_extensions` can be used in `.pyi` files without it actually being available at runtime, I thought that it wouldn't be unlikely that this import would raise questions. So that's what I want to prevent with this.

And For those interested in python-typing that found this comment by accident could maybe find this little-known fact interesting. 
But perhaps it's common knowledge, and I'm just late to the game 🤷🏻. 

The import might also cause onlookers to conclude that `typing_extensions` is a required dependency, and try to use it at runtime in the numpy codebase. So this comment could help prevent situations like that.

Either way, I don't see any harm in leaving it here, as it directly applies to the import below, and is a brief and IMO clear explanation of a non-obvious, but useful, typing fact.

And for those that want to know more; they'll easily be able to find this PR through git blame, which elaborates on it.